src/Features/Core/Portable/FindUsages/DefinitionItemFactory.cs

```diff
@@ -151,75 +116,108 @@ internal static class DefinitionItemExtensions
+                    .Add(DefinitionItem.MetadataSymbolOriginatingProjectIdDebugName, originatingProjectId.DebugName ?? "");
             }
 
-            if (sourceLocations.IsEmpty)
+            if (sourceLocations.IsEmpty && metadataLocations.IsEmpty)
             {
                 // If we got no definition locations, then create a sentinel one
                 // that we can display but which will not allow navigation.
                 return DefinitionItem.CreateNonNavigableItem(
                     tags, displayParts,
-                    DefinitionItem.GetOriginationParts(definition),
+                    nameDisplayParts,
+                    metadataLocations,
                     properties, displayIfNoReferences);
             }
 
             var displayableProperties = AbstractReferenceFinder.GetAdditionalFindUsagesProperties(definition);
 
             return DefinitionItem.Create(
-                tags, displayParts, sourceLocations, classifiedSpans,
+                tags, displayParts, sourceLocations, classifiedSpans, metadataLocations,
                 nameDisplayParts, properties, displayableProperties, displayIfNoReferences);
         }
 
-        private static ImmutableArray<DocumentSpan> TryGetSourceLocations(ISymbol definition, Solution solution, ImmutableArray<Location> locations, bool includeHiddenLocations)
+        internal static ImmutableArray<AssemblyLocation> GetMetadataLocations(ISymbol definition, Solution solution, out ProjectId? originatingProjectId)
         {
-            // If it's a namespace, don't create any normal location.  Namespaces
-            // come from many different sources, but we'll only show a single 
-            // root definition node for it.  That node won't be navigable.
-            if (definition.Kind == SymbolKind.Namespace)
+            originatingProjectId = null;
+
+            if (!definition.Locations.Any(static location => location.MetadataModule != null))
             {
                 return [];
             }
 
-            // If it's a namespace, don't create any normal location.  Namespaces
-            // come from many different sources, but we'll only show a single 
-            // root definition node for it.  That node won't be navigable.
-            using var sourceLocations = TemporaryArray<DocumentSpan>.Empty;
+            var assembly = definition as IAssemblySymbol ?? definition.ContainingAssembly;
+            if (assembly != null)
+            {
+                // symbol is defined within a single metadata assembly:
+                var info = solution.CompilationState.GetOriginatingProjectInfo(assembly);
+                Contract.ThrowIfNull(info);
+                Contract.ThrowIfNull(info.ReferencedThrough);
 
-            foreach (var location in locations)
+                originatingProjectId = info.ProjectId;
+                return [new AssemblyLocation(assembly.Identity.Name, assembly.Identity.Version, info.ReferencedThrough.Value.FilePath)];
```

Comment:
is `info.ReferencedThrough.Value` always ok?

Reply:
Yes, `Contract.ThrowIfNull(info.ReferencedThrough);` above

Reply:
fair.  put another way, how are we sure that `solution.CompilationState.GetOriginatingProjectInfo(assembly)` will work? :)